packages/@glimmer/runtime/lib/compiled/blocks.ts

```diff
@@ -1,19 +1,10 @@
-import { Slice } from '../opcodes';
-
 export class CompiledBlock {
-  public start: number;
-  public end: number;
-  public slice: Slice;
-
-  constructor(slice: Slice) {
-    this.start = slice[0];
-    this.end = slice[1];
-    this.slice = slice;
+  constructor(public start: number, public end: number) {
   }
 }
 
 export class CompiledProgram extends CompiledBlock {
```

Comment:
I think you could just use `CompiledBlock` in the places `CompiledProgram` is being used. It delegates everything to `CompiledBlock`.